cloud/google/clusteractuator.go

```diff
@@ -94,19 +108,27 @@ func (gce *GCEClusterClient) Reconcile(cluster *clusterv1.Cluster) error {
 	if err != nil {
 		glog.Warningf("Error creating firewall rule for core api server traffic: %v", err)
 	}
+
+	// If any resources had to be created for this cluster, emit a created event.
+	if createdInternalRule || createdApiRule {
+		gce.eventRecorder.Eventf(cluster, corev1.EventTypeNormal, "Created", "Created Cluster %v", cluster.Name)
```

Comment:
should it be a separate event for each firewall rule?

Reply:
I don't want to emit more than 1 event for the cluster as a whole because that doesn't really follow the pattern established by existing events in the system. Especially since I expect this function to create more resources in the future, it could end up emitting quite a few events if we do one per resource. If either of the firewall rules fails to create, a failure event will show up, which seems sufficient to help with debugging. What do you think?

Reply:
The scenario I am worried about is that the user sees a failure due to firewall, not knowing which firewall, they will end up going to the logs, which seems redundant if we can provide that information in the event.
Since this is in regards to Create in particular, I am fine with grouping them together.
I am just trying to understand what is the approach that we are taking with events. It might be good to document it for future maintainers.
For example, create/delete per resource grouping, failures per incident?

Reply:
Personally I think it's ok to have people look at logs or resource status errors to figure out more information about errors. Events aren't usually detailed [per the style guide](https://github.com/kubernetes/community/blob/master/contributors/devel/event-style-guide.md). I think it should only emit one warning event per creation too, it won't right now because the failure of one firewall rule to create will not stop execution of this function. But maybe that decision should be revisited..